x/stake/msg.go

```diff
@@ -45,11 +46,9 @@ func (msg MsgDeclareCandidacy) GetSigners() []sdk.Address { return []sdk.Address
 
 // get the bytes for the message signer to sign on
 func (msg MsgDeclareCandidacy) GetSignBytes() []byte {
-	b, err := json.Marshal(msg)
-	if err != nil {
-		panic(err)
-	}
-	return b
+	cdc := wire.NewCodec()
```

Comment:
Do we need to create a new codec each call?

Reply:
struggling with how to use a basic cdc without needing to pass them all around. But, correct good point this can just be defined once here. I just don't like having to many cdc's lying around all over the place - have a solution in mind here though, shouldn't be too bad if we define it globally (with an `init` function too) right above this message.

Reply:
A singleton for the file (or even just for all calls of this function) seems fine to me.

Reply:
yeah that's the plan - I just want the cdc definitions to be as consolidated as possible - it's more just stylistic clarity I'm thinking of